pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -96,11 +97,47 @@ func createCombination(name string, value string, combination Params) Params {
 
 // CountCombinations returns the count of combinations of Parameters generated from the Matrix in PipelineTask.
 func (m *Matrix) CountCombinations() int {
+	// Explicit combinations in the Matrix
+	if m.hasInclude() && !m.hasParams() {
+		return len(m.Include)
+	}
```

Comment:
don't think this is necessary
```suggestion
```

Reply:
@jerop Correct me if I'm wrong but I believe this is necessary if there are only Matrix Include params for them to be dealt with separately to only generate the explicit combinations as opposed to generating all combinations.

Reply:
I think it's not necessary because in that scenario: 
- countMatrixParamsCombinations() == 0
- countNewCombinationsFromInclude() == len(m.Include)
So it ends up being the same thing? Or maybe you can add that check in countNewCombinationsFromInclude() to make it faster?

Reply:
Oh I see, it was failing before when I just removed the block of code. I think adding a check in `countNewCombinationsFromInclude` makes sense